src/modules/BootStorage.rb

```diff
@@ -139,22 +139,27 @@ def underlaying_devices(dev)
           end
         elsif disk["type"] == :CT_LVM
           # not happy with this usage of || but target map do not need to have it defined
-          res = (disk["devices"] || []) + (disk["devices_add"] || [])
+          res = devices_on(disk)
+          # skip lvm on partiotionless disks as it cannot be used, see bnc#980529
+          res.delete_if { |d| tm[d] }
           res.map! { |r| Yast::Storage.GetDiskPartition(r)["disk"] }
         end
-      else
+      else # partition
```

Comment:
Additionally, please, move comments for if/else `above` the if/else so they start at the same level (indentation) of code